### PR TITLE
update artifactId groupId and version for springboot module

### PR DESCRIPTION
**Pull request by wujack778**
Update artifactId, groupId and version 
fix [issue11374](https://github.com/azure/azure-sdk-for-java/issues/11374) and [issue11153](https://github.com/azure/azure-sdk-for-java/issues/11153)

### CODE REVIEW

**Comment by chenrujun** [comment]
Hi, @wujack778 , please resolve the conflicts. 🙏

**Comment by wujack778** [comment]
all right
________________________________

File: sdk/boms/azure-spring-boot-bom/pom.xml [hunk 1]
```diff
@@ -57,44 +57,44 @@
...
         <artifactId>azure-spring-boot-starter</artifactId>
         <version>${azure.spring.boot.jms.integration.version}</version>
       </dependency>
       <dependency>
-        <groupId>com.microsoft.azure</groupId>
-        <artifactId>azure-active-directory-spring-boot-starter</artifactId>
-        <version>${azure.spring.boot.integration.version}</version>
+        <groupId>com.azure</groupId>
+        <artifactId>azure-spring-boot-starter-active-directory</artifactId>
+        <version>${azure.spring.boot.jms.integration.version}</version>
```
Comment (chenrujun) [inline sdk/boms/azure-spring-boot-bom/pom.xml:72]: Do we need to add `jms` here?

File: sdk/boms/azure-spring-boot-bom/pom.xml [hunk 2]
```diff
@@ -57,44 +57,44 @@
...
       </dependency>
       <dependency>
-        <groupId>com.microsoft.azure</groupId>
-        <artifactId>azure-active-directory-b2c-spring-boot-starter</artifactId>
-        <version>${azure.spring.boot.integration.version}</version>
+        <groupId>com.azure</groupId>
+        <artifactId>azure-spring-boot-starter-active-directory-b2c</artifactId>
+        <version>${azure.spring.boot.jms.integration.version}</version>
```
Comment (chenrujun) [inline sdk/boms/azure-spring-boot-bom/pom.xml:77]: Same here.

File: sdk/boms/azure-spring-boot-bom/pom.xml [hunk 3]
```diff
@@ -57,44 +57,44 @@
...
       </dependency>
       <dependency>
-        <groupId>com.microsoft.azure</groupId>
-        <artifactId>azure-cosmosdb-spring-boot-starter</artifactId>
-        <version>${azure.spring.boot.integration.version}</version>
+        <groupId>com.azure</groupId>
+        <artifactId>azure-spring-boot-starter-cosmosdb</artifactId>
+        <version>${azure.spring.boot.jms.integration.version}</version>
```
Comment (chenrujun) [inline sdk/boms/azure-spring-boot-bom/pom.xml:82]: Same here.

File: sdk/boms/azure-spring-boot-bom/pom.xml [hunk 4]
```diff
@@ -57,44 +57,44 @@
...
       </dependency>
 
       <dependency>
-        <groupId>com.microsoft.azure</groupId>
-        <artifactId>azure-keyvault-secrets-spring-boot-starter</artifactId>
-        <version>${azure.spring.boot.integration.version}</version>
+        <groupId>com.azure</groupId>
+        <artifactId>azure-spring-boot-starter-keyvault-secrets</artifactId>
+        <version>${azure.spring.boot.jms.integration.version}</version>
```
Comment (chenrujun) [inline sdk/boms/azure-spring-boot-bom/pom.xml:88]: Same here.

File: sdk/boms/azure-spring-boot-bom/pom.xml [hunk 5]
```diff
@@ -57,44 +57,44 @@
...
       </dependency>
       <dependency>
-        <groupId>com.microsoft.azure</groupId>
-        <artifactId>azure-spring-boot-metrics-starter</artifactId>
-        <version>${azure.spring.boot.integration.version}</version>
+        <groupId>com.azure</groupId>
+        <artifactId>azure-spring-boot-starter-metrics</artifactId>
+        <version>${azure.spring.boot.jms.integration.version}</version>
```
Comment (chenrujun) [inline sdk/boms/azure-spring-boot-bom/pom.xml:93]: Same here.

**Review by chenrujun — CHANGES_REQUESTED** [review]

File: sdk/boms/azure-spring-boot-bom/pom.xml (hunk 1 above)
Comment (wujack778) [reply sdk/boms/azure-spring-boot-bom/pom.xml:72]: you are right

File: eng/versioning/version_client.txt [hunk 6]
```diff
@@ -88,15 +88,15 @@ com.azure.resourcemanager:azure-resourcemanager-sql;2.0.0-beta.4;2.0.0-beta.5
...
-com.microsoft.azure:azure-spring-boot;2.3.5;2.4.0-beta.1
+com.azure:azure-spring-boot-starter-active-directory-b2c;3.0.0-beta.1;3.0.0-beta.1
+com.azure:azure-spring-boot-starter-active-directory;3.0.0-beta.1;3.0.0-beta.1
+com.azure:azure-spring-boot-starter-cosmosdb;3.0.0-beta.1;3.0.0-beta.1
+com.azure:azure-spring-boot-starter-data-gremlin;3.0.0-beta.1;3.0.0-beta.1
+com.azure:azure-spring-boot-starter-keyvault-secrets;3.0.0-beta.1;3.0.0-beta.1
+com.azure:azure-spring-boot-starter-servicebus-jms;3.0.0-beta.1;3.0.0-beta.1
+com.azure:azure-spring-boot-starter-metrics;3.0.0-beta.1;3.0.0-beta.1
+com.azure:azure-spring-boot-starter;3.0.0-beta.1;3.0.0-beta.1
+com.azure:azure-spring-boot;3.0.0-beta.1;3.0.0-beta.1
```
Comment (saragluna) [inline eng/versioning/version_client.txt:99]: Please sort these lines.

File: eng/jacoco-test-coverage/pom.xml [hunk 7]
```diff
@@ -328,49 +328,49 @@
...
+      <groupId>com.azure</groupId>
+      <artifactId>azure-spring-boot-starter-active-directory-b2c</artifactId>
+      <version>3.0.0-beta.1</version> <!-- {x-version-update;com.azure:azure-spring-boot-starter-active-directory-b2c;current} -->
     </dependency>
     <dependency>
-      <groupId>com.microsoft.azure</groupId>
-      <artifactId>azure-cosmosdb-spring-boot-starter</artifactId>
-      <version>2.4.0-beta.1</version> <!-- {x-version-update;com.microsoft.azure:azure-cosmosdb-spring-boot-starter;current} -->
+      <groupId>com.azure</groupId>
+      <artifactId>azure-spring-boot-starter-cosmosdb</artifactId>
```
Comment (saragluna) [inline eng/jacoco-test-coverage/pom.xml:367]: Cosmos has changed their brand name from cosmosdb to cosmos, we might should follow this change.

File: sdk/spring/azure-spring-boot-starter-active-directory-b2c/pom.xml [hunk 8]
```diff
@@ -98,7 +98,7 @@
           <rules>
             <bannedDependencies>
               <includes>
-                <include>com.microsoft.azure:azure-spring-boot:[2.4.0-beta.1]</include> <!-- {x-include-update;com.microsoft.azure:azure-spring-boot;current} -->
+                <include>com.azure:azure-spring-boot:[3.0.0-beta.1]</include> <!-- {x-include-update;com.azure:azure-spring-boot;current} -->
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-starter-active-directory-b2c/pom.xml:101]: We don't need this anymore, the parent pom already adds exclusions for com.azure:*.

File: sdk/spring/azure-spring-boot-starter-active-directory/pom.xml [hunk 9]
```diff
@@ -82,7 +82,7 @@
           <rules>
             <bannedDependencies>
               <includes>
-                <include>com.microsoft.azure:azure-spring-boot:[2.4.0-beta.1]</include> <!-- {x-include-update;com.microsoft.azure:azure-spring-boot;current} --> -->
+                <include>com.azure:azure-spring-boot:[3.0.0-beta.1]</include> <!-- {x-include-update;com.azure:azure-spring-boot;current} --> -->
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-starter-active-directory/pom.xml:85]: same here.

File: sdk/spring/azure-spring-boot-starter-cosmosdb/pom.xml [hunk 10]
```diff
@@ -52,7 +52,7 @@
           <rules>
             <bannedDependencies>
               <includes>
-                <include>com.microsoft.azure:azure-spring-boot:[2.4.0-beta.1]</include> <!-- {x-include-update;com.microsoft.azure:azure-spring-boot;current} -->
+                <include>com.azure:azure-spring-boot:[3.0.0-beta.1]</include> <!-- {x-include-update;com.azure:azure-spring-boot;current} -->
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-starter-cosmosdb/pom.xml:55]: same here.

File: sdk/spring/azure-spring-boot-starter-data-gremlin/pom.xml [hunk 11]
```diff
@@ -56,7 +56,7 @@
           <rules>
             <bannedDependencies>
               <includes>
-                <include>com.microsoft.azure:azure-spring-boot:[2.4.0-beta.1]</include> <!-- {x-include-update;com.microsoft.azure:azure-spring-boot;current} -->
+                <include>com.azure:azure-spring-boot:[3.0.0-beta.1]</include> <!-- {x-include-update;com.azure:azure-spring-boot;current} -->
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-starter-data-gremlin/pom.xml:59]: same here.

File: sdk/spring/azure-spring-boot-starter-keyvault-secrets/pom.xml [hunk 12]
```diff
@@ -51,7 +51,7 @@
           <rules>
             <bannedDependencies>
               <includes>
-                <include>com.microsoft.azure:azure-spring-boot:[2.4.0-beta.1]</include> <!-- {x-include-update;com.microsoft.azure:azure-spring-boot;current} -->
+                <include>com.azure:azure-spring-boot:[3.0.0-beta.1]</include> <!-- {x-include-update;com.azure:azure-spring-boot;current} -->
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-starter-keyvault-secrets/pom.xml:54]: same here.

File: sdk/spring/azure-spring-boot-starter-keyvault-secrets/pom.xml [hunk 13]
```diff
@@ -51,7 +51,7 @@
...
                 <include>com.azure:azure-security-keyvault-secrets:[4.2.1]</include> <!-- {x-include-update;com.azure:azure-security-keyvault-secrets;dependency} -->
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-starter-keyvault-secrets/pom.xml:55]: maybe this is redundant too.

File: sdk/spring/azure-spring-boot-starter-metrics/pom.xml [hunk 14]
```diff
@@ -57,7 +57,7 @@
           <rules>
             <bannedDependencies>
               <includes>
-                <include>com.microsoft.azure:azure-spring-boot:[2.4.0-beta.1]</include> <!-- {x-include-update;com.microsoft.azure:azure-spring-boot;current} -->
+                <include>com.azure:azure-spring-boot:[3.0.0-beta.1]</include> <!-- {x-include-update;com.azure:azure-spring-boot;current} -->
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-starter-metrics/pom.xml:60]: same here

File: sdk/spring/azure-spring-boot-starter-servicebus-jms/pom.xml [hunk 15]
```diff
@@ -97,7 +97,7 @@
                 <include>io.netty:netty-transport-native-epoll:[4.1.51.Final]</include> <!-- {x-include-update;io.netty:netty-transport-native-epoll;external_dependency} -->
                 <include>io.netty:netty-transport-native-kqueue:[4.1.51.Final]</include> <!-- {x-include-update;io.netty:netty-transport-native-kqueue;external_dependency} -->
                 <include>io.netty:netty-codec-http:[4.1.51.Final]</include> <!-- {x-include-update;io.netty:netty-codec-http;external_dependency} -->
-                <include>com.microsoft.azure:azure-spring-boot:[2.4.0-beta.1]</include> <!-- {x-include-update;com.microsoft.azure:azure-spring-boot;current} -->
+                <include>com.azure:azure-spring-boot:[3.0.0-beta.1]</include> <!-- {x-include-update;com.azure:azure-spring-boot;current} -->
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-starter-servicebus-jms/pom.xml:100]: same here

File: sdk/spring/azure-spring-boot-starter/pom.xml [hunk 16]
```diff
@@ -47,7 +47,7 @@
           <rules>
             <bannedDependencies>
               <includes>
-                <include>com.microsoft.azure:azure-spring-boot:[2.4.0-beta.1]</include> <!-- {x-include-update;com.microsoft.azure:azure-spring-boot;current} -->
+                <include>com.azure:azure-spring-boot:[3.0.0-beta.1]</include> <!-- {x-include-update;com.azure:azure-spring-boot;current} -->
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-starter/pom.xml:50]: same here.

**Review by saragluna — CHANGES_REQUESTED** [review]

**Comment by wujack778** [comment]
Because this branch combines the modification of packagename and the modification of artifactID, I will open a new branch for the modification artifactID and readme [link](https://github.com/Azure/azure-sdk-for-java/pull/15310#issue-488442478)

**Pull request closed without merging**